core/src/view_functions/mod.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright 2019-2024 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+//! Encode View Function payloads, decode the associated values returned from them, and validate
+//! static View Function payloads.
+
+pub mod payload;
+
+use crate::error::{Error, MetadataError};
+use crate::metadata::{DecodeWithMetadata, Metadata};
+use alloc::vec::Vec;
+use payload::Payload;
+
+/// Run the validation logic against some View Function payload you'd like to use. Returns `Ok(())`
+/// if the payload is valid (or if it's not possible to check since the payload has no validation hash).
+/// Return an error if the payload was not valid or something went wrong trying to validate it (ie
+/// the View Function in question do not exist at all)
+pub fn validate<P: Payload>(payload: &P, metadata: &Metadata) -> Result<(), Error> {
+    let Some(static_hash) = payload.validation_hash() else {
+        return Ok(());
+    };
+
+    let view_function = metadata
+        .view_function_by_query_id(payload.query_id())
+        .ok_or_else(|| MetadataError::ViewFunctionNotFound(*payload.query_id()))?;
+    if static_hash != view_function.hash() {
+        return Err(MetadataError::IncompatibleCodegen.into());
+    }
+
+    Ok(())
+}
+
+/// Return the name of the Runtime API call which can execute
+pub fn call_name() -> &'static str {
```

Comment:
dq: We plan to extend this in the future by inspecting the metadata and therefore we picked a plain fn call instead of const VIEW_FN: ..?

Reply:
Made it a const!